### PR TITLE
Resolve all annotation values in a uniform way

### PR DESCRIPTION
**Pull request by hemikak**

### CODE REVIEW

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java [hunk 1]
```diff
@@ -299,63 +299,86 @@ public static String resolveValue(String variable) throws KubernetesPluginExcept
...
+     * @return Convert string.
      */
-    public static Map<String, String> getMap(List<BLangRecordLiteral.BLangRecordKeyValue> keyValues)
-            throws KubernetesPluginException {
-        Map<String, String> map = new LinkedHashMap<>();
-        if (keyValues != null) {
-            for (BLangRecordLiteral.BLangRecordKeyValue keyValue : keyValues) {
-                map.put(keyValue.getKey().toString(), resolveValue(keyValue.getValue().toString()));
+    public static List<String> getArray(BLangExpression expr) throws KubernetesPluginException {
+        if (expr instanceof BLangArrayLiteral) {
```
Comment (NatashaWso2) [inline kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: Here without using 'instanceof' we can check the node kind. 

```java
if (expr.getKind() == NodeKind.ARRAY_LITERAL_EXPR) {
```

Comment (NatashaWso2) [reply kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: We can also invert the condition and return the error first.

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java [hunk 2]
```diff
@@ -299,63 +299,86 @@ public static String resolveValue(String variable) throws KubernetesPluginExcept
...
+     * @return Map of key values.
+     * @throws KubernetesPluginException When the expression cannot be parsed.
      */
-    public static DeploymentBuildExtension parseBuildExtension(BLangExpression buildExtensionValue)
-            throws KubernetesPluginException {
-        if (buildExtensionValue instanceof BLangSimpleVarRef || buildExtensionValue instanceof BLangLiteral) {
-            if ("openshift".equals(getStringValue(buildExtensionValue))) {
-                return new OpenShiftBuildExtensionModel();
+    public static Map<String, String> getMap(BLangExpression expr) throws KubernetesPluginException {
+        if (expr instanceof BLangRecordLiteral) {
```
Comment (NatashaWso2) [inline kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: Here without using 'instanceof' we can check the node kind. We can also invert the if condition and return the error soon.

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java [hunk 3]
```diff
@@ -369,19 +392,32 @@ public static String getStringValue(BLangExpression expr) throws KubernetesPlugi
...
+     * @return Build extension model.
+     * @throws KubernetesPluginException When an unknown extension is found.
      */
-    public static Set<String> getArray(BLangArrayLiteral bArrayLiteral) throws KubernetesPluginException {
-        Set<String> scopeSet = new LinkedHashSet<>();
-        for (ExpressionNode bLangExpression : bArrayLiteral.getExpressions()) {
-            scopeSet.add(resolveValue(bLangExpression.toString()));
+    public static DeploymentBuildExtension parseBuildExtension(BLangExpression buildExtensionValue)
+            throws KubernetesPluginException {
+        if (buildExtensionValue instanceof BLangSimpleVarRef || buildExtensionValue instanceof BLangLiteral) {
```
Comment (NatashaWso2) [inline kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: Here without using 'instanceof' we can check the node kind.

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java [hunk 4]
```diff
@@ -369,19 +392,32 @@ public static String getStringValue(BLangExpression expr) throws KubernetesPlugi
...
+            if ("openshift".equals(getStringValue(buildExtensionValue))) {
+                return new OpenShiftBuildExtensionModel();
+            }
+        } else {
+            if (buildExtensionValue instanceof BLangRecordLiteral) {
```
Comment (NatashaWso2) [inline kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: Here without using 'instanceof' we can check the node kind.

Comment (hemikak) [reply kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: Fixed with https://github.com/ballerinax/kubernetes/pull/301/commits/66ec19a0fb6d97a36b6121d8c27adf89e2956468

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java (hunk 2 above)
Comment (hemikak) [reply kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: Fixed with https://github.com/ballerinax/kubernetes/pull/301/commits/66ec19a0fb6d97a36b6121d8c27adf89e2956468

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java (hunk 1 above)
Comment (hemikak) [reply kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: Fixed with https://github.com/ballerinax/kubernetes/pull/301/commits/66ec19a0fb6d97a36b6121d8c27adf89e2956468

File: kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java (hunk 3 above)
Comment (hemikak) [reply kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java]: This cannot be done as I am expecting a parent type `BLangSimpleVarRef` or `BLangLiteral`.

**Pull request merged**